### ISSUE TITLE
NPM package azure-streamanalytics-cicd nested ARM templates

### ISSUE THREAD

**Issue by assaf-grth** [microsoft/vscode-asa#76]
We are using the npm package and receive this error during Azure Devops CICD build:
[error]Deployment template validation failed: 'The number of template input parameters limit exceeded. Limit: '256' and actual: '264'. Please see [https://aka.ms/arm-template/#parameters for usage details.'.](https://aka.ms/arm-template/#parameters%C2%A0for%C2%A0usage%C2%A0details.%27.)

The stream job that the process creates contains a lot of outputs which causes a very large amount of template parameters, breaking the build

Is there a way of mitigating such issue using this tool?
Maybe some additional configuration that can be provided for its execution?

**Comment by assaf-grth** [comment]
Thanks for the update @pengyuli-ms,
We might be able to wait,
Can you share a more exact date for the release handling this issue?

**Comment by assaf-grth** [comment]
Hi @pengyuli-ms 
Any updates regarding the fix? Is there a more accurate due date?

**Comment by pengyuli-ms** [comment]
Hi @assaf-grth, we're sorry to say that this feature might be delayed to next month, due to recent severe COVID situation in our region.

**Comment by assaf-grth** [comment]
Hope everyone are feeling well..
So the next due date is around middle of next month?
Maybe there is a preview version we could start use?

**Comment by pengyuli-ms** [comment]
Yes, next month release will include this feature for sure, and sorry that we didn't have a `preview` version available for now, thanks for your understanding!

**Comment by assaf-grth** [comment]
Hi @pengyuli-ms
Any updates regarding the fix?

**Comment by pengyuli-ms** [comment]
Hi @assaf-grth , we've completed the code development phase for this fix (actually we did a deep refactor to fix this bug and some other issues).
Currently we're under test phase, and will release it once done, before Feb. 10th.

**Comment by assaf-grth** [comment]
Hi @pengyuli-ms thanks for the update and sorry for the late response

I checked the documentation and didn't see any mention of templates nesting or recommendation for case of more than 256 parameters

Should I use the command line with the addition of `--v2` and the nesting support will kick in automatically? Or should I prepare differently to using the new package version?

**Comment by pengyuli-ms** [comment]
That's correct @assaf-grth , just need to add the `--v2` option.

**Comment by ruiminwang** [comment]
Hi @assaf-grth, does the new npm package with `--v2`  parameter resolve your problem? We'll close this issue if your problem has been resolved.

**Comment by assaf-grth** [comment]
Hi @pengyuli-ms @ruiminwang 
We had a problem with using the new version of the package
We use a project based on VS (with asaproj extension)

When executing the package with this project type, the execution fails with this error
```
[Error] Failed to generate ARM template for project 'ASA_Job/ASA_Job.asaproj'. Exception details: System.Exception: Failed to load project '..some path...\ASA_Job.asaproj': Unexpected character encountered while parsing value: <. Path '', line 0, position 0.. 
```

It looks like the new version expects to receive only json files (the VS code project type) and we use the VS project version

Can you advice?

Thanks,
Assaf

**Comment by pengyuli-ms** [comment]
Actually, the VS extension has been deprecated, and only the VS Code extension is currently supported for new features and bug fixes. 

Therefore, we recommend using our VS Code extension instead. Migrating to the VS Code extension is quite simple. You just need to install it and export your ASA job from the Portal.

Here is a step-by-step guidance: https://learn.microsoft.com/en-us/azure/stream-analytics/migrate-to-vscode#option-1--export-from-the-azure-portal

**Comment by assaf-grth** [comment]
Thanks @pengyuli-ms for the update,
Is there an official statement about this deprecation?
I only found this message [here](https://learn.microsoft.com/en-us/azure/stream-analytics/stream-analytics-tools-for-visual-studio-install) which doesn't mention deprecation directly

![image](https://user-images.githubusercontent.com/52287325/228749790-08f13d82-5636-4503-ad88-82ce4cbec187.png)

**Comment by pengyuli-ms** [comment]
Sorry for any confusion caused. We will be making an official announcement about the deprecation soon.
Thank you for pointing this out to us!
